code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+	update_use_power(POWER_USE_IDLE)
+	do_effects(jumpdist)
+	jumping = FALSE
+	update_icon()
+	chargepercent = 0
+	charge_started = 0
+
+//Handles all the effects of the jump.
+/obj/machinery/ftl_shunt/core/proc/do_effects(var/distance) //If we're jumping too far, have some !!FUN!! with people and ship systems.
+	var/shunt_sev
+	switch(distance)
+		if(1 to 3)
+			shunt_sev = SHUNT_SEVERITY_MINOR
+		if(4 to 5)
+			shunt_sev = SHUNT_SEVERITY_MAJOR
+		if(6 to INFINITY)
+			shunt_sev = SHUNT_SEVERITY_CRITICAL
+
+	for(var/mob/living/carbon/human/H in GLOB.living_mob_list_) //Affect mobs, skip synthetics.
+		sound_to(H, 'sound/machines/hyperspace_end.ogg')
+
+		if(!(H.z in ftl_computer.linked.map_z))
+			continue
+
+		handle_spacefloat(H)
+
+		if(isnull(H) || QDELETED(H))
+			continue
+
+		if(H.isSynthetic())
+			continue //We don't affect synthetics.
+
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				to_chat(H, SPAN_NOTICE("You feel your insides flutter about inside of you as you are briefly shunted into an alternate dimension.")) //No major effects.
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_MAJOR)
+				to_chat(H, SPAN_WARNING("You feel your insides twisted inside and out as you are violently shunted between dimensions, and you feel like something is watching you!"))
+				if(prob(25))
+					H.set_hallucination(50, 50)
+				if(prob(15))
+					H.vomit()
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				to_chat(H, SPAN_DANGER("You feel an overwhelming sense of nausea and vertigo wash over you, your instincts screaming that something is wrong!"))
+				if(prob(50))
+					H.set_hallucination(100, 100)
+				if(prob(45))
+					H.vomit()
+				shake_camera(H, 5, 4)
+
+	for(var/obj/machinery/light/L in SSmachines.machinery) //Fuck with and or break lights.
+		if(!(L.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				if(prob(15))
+					L.flicker()
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(35))
+					L.flicker()
+
+	for(var/obj/machinery/power/apc/A in SSmachines.machinery)
+		if(!(A.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(15))
+					A.energy_fail(rand(30, 80))
+				if(prob(10))
+					A.overload_lighting(25)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				if(prob(35))
+					A.energy_fail(rand(60, 150))
+				if(prob(50))
+					A.overload_lighting(50)
+
+/obj/machinery/ftl_shunt/core/proc/handle_spacefloat(var/mob/living/carbon/human/H)
+	if(!H.check_space_footing())
+		 //Flip a coin ...
+		to_chat(H, SPAN_WARNING("Being untethered from a ship entering FTL is a bad idea, but you roll the dice..."))
+		if(prob(50))
+			to_chat(H, SPAN_NOTICE("and win, surviving the energy dancing over your body. Not unharmed, however."))
+			H.apply_damage(300, IRRADIATE, damage_flags = DAM_DISPERSED)
+			return
+		else
+			to_chat(H, SPAN_DANGER("and lose, being ripped apart in a nanosecond by energies beyond comprehension."))
```

Comment:
Could just `gib()` them to be more flexible. Also gibbing humans in space is like fireworks, the limbs go flying away in a pretty flower.